src/algokit/core/compile.py

```diff
@@ -0,0 +1,80 @@
+from collections.abc import Iterator
+
+from algokit.core.proc import run
+from algokit.core.utils import extract_version_triple, find_valid_pipx_command
+
+
+def find_valid_puyapy_command(version: str | None) -> list[str]:
+    return _find_puya_command_at_version(version) if version is not None else _find_puya_command()
+
+
+def _find_puya_command_at_version(version: str) -> list[str]:
+    """
+    Find puya command with a specific version.
+    If the puya version isn't installed, install it with pipx run.
+    """
+    for puyapy_command in _get_candidates_puyapy_commands():
+        try:
+            puyapy_version_result = run([*puyapy_command, "--version"])
```

Comment:
Puya doesn't support `--version` yet, we catch it here
```
if puyapy_version_result.exit_code == 0
```
That means, if the user requests for a specific puya version, `pipx run` is always used, until puya support `--version` option.

Reply:
FYI this has been added, it just hasn't been released to PyPi yet https://github.com/algorandfoundation/puya/pull/110